File: solutions/Python/11.ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area: int = 0
        left: int = 0
        right: int = len(height) - 1

        while left < right:
            area: int = (right - left) * min(height[left], height[right])
            max_area = max(max_area, area)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_area

File: solutions/TypeScript/11.ContainerWithMostWater.ts
function maxArea(height: number[]): number {
  let maxArea: number = 0;
  let left: number = 0;
  let right: number = height.length - 1;

  while (left < right) {
    let area: number = (right - left) * Math.min(height[left], height[right]);
    maxArea = Math.max(maxArea, area);

    height[left] < height[right] ? left++ : right--;
  }

  return maxArea;
}
